Introduce vlib implementation module

rgrinberg opened the pull request:
In the old code, the fact that our virtual lib is local is assumed in too many places. I attempt to make this problem easier to address by introducing a `Vimpl` module that should encapsulate all that is necessary for transparently generating implementation rules; regardless of where/how the virtual lib lives.

Note that I had to split the ocamldep module to avoid cycles.

diml commented:
I made the assumption that the new modules where code that was moved without being modified and so didn't read them in detail. Let me know if that's not the case.

rgrinberg commented:
That's indeed the case. I just moved more logic to the self contained `Vimpl` module. This required me to split more stuff to break some cycles.

Why did the CI fail? I added the no changes label.
